src/gen9_mfd.c

```diff
@@ -349,6 +353,66 @@ gen9_hcpd_ind_obj_base_addr_state(VADriverContextP ctx,
     ADVANCE_BCS_BATCH(batch);
 }
 
+static void
+gen10_hcpd_ind_obj_base_addr_state(VADriverContextP ctx,
+                                   dri_bo *slice_data_bo,
+                                   struct gen9_hcpd_context *gen9_hcpd_context)
+{
+    struct intel_batchbuffer *batch = gen9_hcpd_context->base.batch;
+
+    BEGIN_BCS_BATCH(batch, 29);
+
+    OUT_BCS_BATCH(batch, HCP_IND_OBJ_BASE_ADDR_STATE | (29 - 2));
+
+    /* DW1..5 indirect bitstream*/
+    OUT_RELOC64(batch,
+                slice_data_bo,
+                I915_GEM_DOMAIN_INSTRUCTION, 0,
+                0);
+    OUT_BCS_BATCH(batch, 0);
+    OUT_BCS_BATCH(batch, 0);
+    OUT_BCS_BATCH(batch, 0);
```

Comment:
Please add the upper bound check, like as what we did for GEN9.

Reply:
ok, I will add it and remove the unneeded patch